MetaMask/metamask-extension
# refactor: change CurrencyRateController Polling to be UI driven

## **Description**

Currently currency rate polling is started and stopped based on active trusted connections to the `MetaMaskController` (i.e. whenever the wallet ui is opened). In the future we want polling to be tied to the actual components that rely on polled currency rate state. As a transitionary step, this PR moves currency rate polling from `MetaMaskController` into a hook that wraps the root level `<Routes>` component. Nothing changes except that currency rate polling goes from being MetaMaskController connection driven to actually wallet UI driven. Polling will still occur as long at least one wallet UI is open as it currently already does.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/PR?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2083

## **Manual testing steps**

1. Open background console - network tab
3. Filter by `min-api.cryptocompare.com`
4. Open wallet UI
5. See request in background network tab
6. Change network
7. See new request
8. Close wallet UI
9. Open wallet UI right away
10. See no new requests
11. Close wallet UI
12. Wait 3 minutes
13. See no new requests
14. Open wallet UI
15. See new request 
16. Leave wallet UI open
17. Wait 3 minutes
18. See new request

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
